app/models/oclc/record.rb

```diff
@@ -0,0 +1,250 @@
+    end
+
+    def call_number_in_range_for_selector?(selector:)
+      return false unless class_relevant_to_selector?(selector:)
+
+      selector.call_number_ranges.map do |range|
+        next false unless lc_class == range[:class]
+
+        next true if lc_number.between?(range[:low_num], range[:high_num])
+
+        false
+      end.include?(true)
+    end
+
+    def class_relevant_to_selector?(selector:)
+      return true if selector.classes.include?(lc_class)
+
+      false
+    end
+
+    def subject_relevant_to_selector?(selector:)
+      subjects.any? { |subject| subject.downcase.match?(/#{selector.subjects.join('|')}/) }
+    end
+
+    def author
+      # The first available author field
+      auth_field = record.fields(%w[100 110 111]).try(:[], 0)
+      return '' if auth_field.blank?
+
+      auth_tag = auth_field.tag
+      subf_to_skip = auth_subfields_to_skip(auth_tag)
+      targets = auth_field.subfields.reject do |subfield|
+        subf_to_skip.include?(subfield.code)
+      end
+
+      targets.map { |field| scrub_string(field.value) }.join(' ')
+    end
+
+    def description
+      return '' unless record['300']
+
+      record['300'].subfields.map(&:value).join(' ')
+    end
+
+    def f008_pub_place
+      record['008'].value[15..17].strip
+    end
+
+    def format
+      record.leader[6..7]
+    end
+
+    def isbns
+      isbn = []
+      record.fields('020').each do |field|
+        next unless field['a']
+
+        isbn << StdNum::ISBN.normalize(field.value)
+      end
+      isbn.uniq.join(' | ')
+    end
+
+    def lccns
+      lccn = []
+      record.fields('010').each do |field|
+        next unless field['a']
+
+        lccn << StdNum::LCCN.normalize(field.value)
+      end
+      lccn.uniq.join(' | ')
+    end
+
+    def pub_field
+      return record['260'] if record['260']
+
+      record.fields('264').min_by(&:indicator2) if record.fields('264').present?
+    end
+
+    def pub_place
+      return scrub_string(pub_field['a']) if pub_field.try(:[], 'a').present?
+      ''
+    end
+
+    def pub_name
+      return scrub_string(pub_field['b']) if pub_field.try(:[], 'b').present?
+      ''
+    end
+
+    def pub_date
+      return scrub_string(pub_field['c']) if pub_field.try(:[], 'c').present?
+      ''
+    end
+
+    def title
+      title_string = record['245']['a']
+      return '' unless title_string
+
+      scrub_string(title_string)
+    end
+
+    def oclc_id
+      record['001'].value.strip
+    end
+
+    # The LC classification number (050 subfield a) stripped of whitespace
+    def call_number
+      return nil unless record['050']
+
+      targets = record['050'].subfields.select do |subfield|
+        %w[a b].include?(subfield.code)
+      end
+      return nil if targets.empty?
+
+      targets.map(&:value).join(' ')
+    end
+
+    # Only the class from the call_number
+    def lc_class
+      return unless call_number
+
+      call_number.gsub(/^([A-Z]+?)[^A-Z].*$/, '\1')
```

Comment:
Jane and I are a little confused about this regular expression. Could you please explain what this is supposed to do?

Reply:
Also the regexp in lc_number! 😕